includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -639,6 +679,11 @@ private function prepare_wcpay_subscription_data( string $wcpay_customer_id, WC_
 			$data['discounts'] = $discount_items;
 		}
 
+		if ( $this->is_creating_subscription_from_update_payment_method ) {
+			$data['backdate_start_date']  = max( $subscription->get_time( 'last_order_date_created' ), $subscription->get_time( 'last_order_date_paid' ) );
```

Comment:
ah nice. I didn't test that scenario but I can see that being an issue yeah. 

> Maybe I can add $subscription->get_time( 'date_created' ) and/or $subscription->get_time( 'date_paid' ) into this max()

`$subscription->get_time( 'date_paid' )` in this scenario wont work because it just gets the last orders paid date. $subscription->get_time( 'date_created' ) should be fine to use. The other one I was thinking of is `$subscription->get_time( 'start' )`

I think the start date is almost always going to match the created date in normal circumstances. For some context, it was introduced in cases where stores want subscriptions to start in the future not right now. 

Looking at the code in the `WC_Subscription::calculate_next_payment_date()` function, it uses the following date priority when calculating next payment dates:

`max( 'last_order_date_created',  'last_order_date_paid' ) > $start_time`

So using the `$start_time` would be inline with that.